fms/models/llama.py

```diff
@@ -190,7 +190,15 @@ def __init__(
             tie_weights=self.config.tie_heads,
             bias=False,
         )
-        self.shared = self.distributed_strategy.distribute_module(shared)
+
+        # TP does not work with tied weights
+        if (
+            not isinstance(self.distributed_strategy, TensorParallelStrategy)
+            or not self.config.tie_heads
+        ):
+            self.shared = self.distributed_strategy.distribute_module(shared)
+        else:
+            self.shared = shared
```

Comment:
should there be a warning of some sort when this happens? the user may not understand if some checkpoints run out of memory and some don't

Reply:
Sure, I can add a logger warning that this is happening.

Reply:
done